### PR TITLE
Other aircraft

### PR DESCRIPTION
**Pull request by mjh65**
Hi Folke. Here are the updates to show other aircraft as overlays on the map. I've broken the updates into 3 separate commits. Regards, Michael.

### CODE REVIEW

File: src/environment/standalone/StandAloneEnvironment.cpp [hunk 1]
```diff
@@ -120,12 +121,41 @@ void StandAloneEnvironment::reloadMetar() {
...
-    Location res{};
-    res.latitude = 53.8019434;
-    res.longitude = 10.7017287;
-    res.heading = 70;
-    return res;
+unsigned int StandAloneEnvironment::getActiveAircraftCount() {
+    return 4;
+}
+
+Location StandAloneEnvironment::getAircraftLocation(unsigned int id) {
```
Comment (fpw) [inline src/environment/standalone/StandAloneEnvironment.cpp:128]: This is cool, I like the idea of having moving AI aircraft in the standalone version for testing!

File: src/environment/xplane/DataCache.cpp [hunk 2]
```diff
@@ -35,6 +35,44 @@ EnvData DataCache::getData(const std::string& dataRef) {
...
+            char d = '0' + i;
+            locationRefCache.push_back(XPLMFindDataRef((b1 + d + "_lat").c_str()));
+            locationRefCache.push_back(XPLMFindDataRef((b1 + d + "_lon").c_str()));
+            locationRefCache.push_back(XPLMFindDataRef((b1 + d + "_el").c_str()));
+            locationRefCache.push_back(XPLMFindDataRef((b1 + d + "_psi").c_str()));
+        }
+    }
+
+    XPLMDataRef ref = nullptr;
+    int id = plane * 4 + part;
```
Comment (fpw) [inline src/environment/xplane/DataCache.cpp:65]: I see that the aircraft id is sometimes a signed and sometimes an unsigned int, maybe just add a custom type for this? In C++11 that's as simple as adding `using AircraftID = unsigned int;` in some header.

File: src/maps/OverlayedMap.cpp [hunk 3]
```diff
@@ -100,32 +101,37 @@ void OverlayedMap::centerOnWorldPos(double latitude, double longitude) {
...
+void OverlayedMap::centerOnPlane() {
     if (!tileSource->supportsWorldCoords()) {
         return;
     }
 
-    planeLat = latitude;
-    planeLong = longitude;
-    planeHeading = heading;
-    centerOnWorldPos(latitude, longitude);
+    if (planeLocations.size() > 0) {
```
Comment (fpw) [inline src/maps/OverlayedMap.cpp:132]: There's `vector::empty()` to check if a vector is empty, but that's just a minor thing.

**Review by fpw — CHANGES_REQUESTED** [review]
Thanks, looks great! And the thread boundary between X-Plane and AviTab is properly implemented and protected as well. 

I only had a few minor stylistic remarks. They're so minor that I would also accept this PR in its current state - just tell me if you want that.

**Comment by mjh65** [comment]
I've made a few more changes after your review, and after feedback from David and others during a test flight at the weekend.

**Review by fpw — CHANGES_REQUESTED** [review]
Thanks for the changes! The new additions introduced a few other minor things, but after that I'll merge it 🙂

**Review by fpw — APPROVED** [review]
Thanks for considering the suggestions!

**Pull request merged**